src/scripts/main.js

```diff
@@ -1,3 +1,305 @@
 'use strict';
 
-// write code here
+const tbody = document.querySelector('tbody');
+const rows = tbody.getElementsByTagName('tr');
+const titles = document.querySelector('tr');
+const headings = titles.children;
+
+const modeOrder = (value1, value2) =>
+  value1.textContent.localeCompare(value2.textContent);
+
+const replaceExtra = (row) =>
+  row.textContent.replaceAll(',', '').slice(1);
+
+const ascSorter = (index) => [...rows].sort((row1, row2) =>
+  modeOrder(row1.children[index], row2.children[index]));
+
+const descSorter = (index) => [...rows].sort((row1, row2) =>
+  modeOrder(row2.children[index], row1.children[index]));
+
+const ascSorterForSalary = (index) => [...rows].sort((row1, row2) =>
+  replaceExtra(row1.children[index]) - replaceExtra(row2.children[index]));
+
+const descSorterForSalary = (index) => [...rows].sort((row1, row2) =>
+  replaceExtra(row2.children[index]) - replaceExtra(row1.children[index]));
+
+let ascNames = ascSorter(0);
+let ascPosition = ascSorter(1);
+let ascOffice = ascSorter(2);
+let ascAge = ascSorter(3);
+let ascSalary = ascSorterForSalary(4);
+
+let descNames = descSorter(0);
+let descPosition = descSorter(1);
+let descOffice = descSorter(2);
+let descAge = descSorter(3);
+let descSalary = descSorterForSalary(4);
```

Comment:
It's bad practice to create all possible variants and save them (if it'll be a table with 100 columns?)
better to do it "on fly" - just use your sorter functions where you need